src/interface.jl

```diff
@@ -81,13 +117,35 @@ eltype(g::AbstractGraph) = _NI("eltype")
     nv(g)
 
 Return the number of vertices in `g`.
+
+# Examples
+```jldoctest
+julia> using LightGraphs
+
+julia> nv(SimpleGraph(3))
+3
+```
 """
 nv(g::AbstractGraph) = _NI("nv")
 
 """
     ne(g)
 
 Return the number of edges in `g`.
+
+# Examples
+```jldoctest
+julia> using LightGraphs
+
+julia> g = SimpleGraph(2);
+
+julia> add_edge!(g, 1, 2);
+
+julia> add_edge!(g, 1, 1);
```

Comment:
where possible, we want to discourage using self-loops. Several algorithms will break on self-loops, and they have a very specific use case for a couple of high-profile users.

Can we make this a `PathGraph(3)` to show `ne(g) == 2` ?